recipes-devtools/firmware-update-package/files/iot2050-generate-fwu-tarball.sh

```diff
@@ -9,16 +9,43 @@
 # COPYING.MIT file in the top-level directory.
 #
 
-update_json()
-{
+update_json() {
     version_letter="$(echo $2 | head -c 1)"
-    sed -i '/"version": ".*"/s/"V.*"/"'$2'"/g' $1
-    sed -i '/"description": ".*"/s/V.*["$]/'$2\"'/g' $1
-    sed -i '/"min_version": ".*"/s/"V/"'$version_letter'/g' $1
+
+    sed -i '/"version": ".*"/s|"V.*"|"'$2'"|g' $1
+    sed -i '/"description": ".*"/s|V.*["$]|'$2\"'|g' $1
+    sed -i '/"min_version": ".*"/s|"V|"'$version_letter'|g' $1
+
+    # if the SUGGEST_PRESERVED_UBOOT_ENV is not empty, update the
+    # suggest_preserved_uboot_env in the update.conf.json
+    if [ ! -z "$SUGGEST_PRESERVED_UBOOT_ENV" ]; then
```

Comment:
Add a new yml template, then pass the SCM value via this yaml.

Reply:
As we discussed, remove this undue feature.